mdcli/validate.go

```diff
@@ -27,20 +27,10 @@ func Validate(opts *CommandOptions) (int, error) {
 
 	valErr, err := mdProcessor.Validate(cli)
 	if err != nil {
+		fmt.Fprintf(opts.Stderr, "Error: %s\nReason: %s\n", valErr.Error, valErr.Message)
 		return 0, err
```

Comment:
I think you flipped the logic here.  Now the error condition returns `0` and the non-error condition returns `1` which will result in the shell exit code, so we need to swap the return values as well.

Reply:
I think that it was wrong before (see line 15), but I'll fix it :)

Reply:
Interesting, yeah, there were some odd return values before.  I didnt notice before but this structure is "odd" from a Go perspective.  Normally when a function returns `foo, err := ...` then if `err` is not nil it should be assumed that `foo` is invalid and not usable (likely a `nil` pointer).  I think previously `err` would be non-nil only in the "invalid http request/response", so transport, or auth errors where we would fail to parse the response from the spinnaker validation api.  So I think we need to keep that consistent.  If `err == nil` we have well-formed response from the validation api (which may still mean invalid delivery config ) and if `err != nil` we dont have a valid response (auth error etc).

Reply:
if `valErr != nil` then `err != nil`
I pushed a fix, lmk if it makes more sense